vehicle/mkvi/software/bms/tasks/openwire.c

```diff
@@ -4,19 +4,115 @@
 #include "vehicle/mkvi/software/bms/bms_config.h"
 #include "vehicle/mkvi/software/bms/can_api.h"
 #include "vehicle/mkvi/software/bms/utils/fault.h"
-
+#include <string.h>
 
 #define NO_OPEN_WIRES (65535)
+#define NUM_CV_REG    (4)
+#define NUM_CHANNELS  (18)
+
+/*
+Reads and parses the LTC681x cell voltage registers.
+The function is used to read the parsed Cell voltages codes of the LTC681x.
+This function will send the requested read commands parse the data
+and store the cell voltages in c_codes variable.
+*/
+uint8_t LTC681x_rdcv_trunc(
+    uint8_t reg, // Controls which cell voltage register is read back.
+    uint8_t total_ic // The number of ICs in the system
+) {
+    int8_t pec_error = 0;
+    uint8_t cell_data_buff[NUM_RX_BYT * total_ic];
+    memset(cell_data_buff, 0x00, NUM_RX_BYT * total_ic);
+
+    if (reg == 0) {
+        for (uint8_t cell_reg = 1; cell_reg < NUM_CV_REG + 1;
+             cell_reg++) // Executes once for each of the LTC681x cell voltage
+                         // registers
+        {
+            LTC681x_rdcv_reg(cell_reg, total_ic, cell_data_buff);
+            for (int current_ic = 0; current_ic < total_ic; current_ic++) {
+                pec_error
+                    = pec_error
+                      + parse_cells(
+                          current_ic, cell_reg, cell_data_buff,
+                          &cell_data.cells[current_ic].cells.c_codes[0],
+                          &cell_data.cells[current_ic].cells.pec_match[0]);
+            }
+        }
+    }
+
+    else {
```

Comment:
u can probably delete this arm of the if. since the function is only called in one place and we call it for all registers